src/view3d.cpp

```diff
@@ -271,23 +403,39 @@ void View3D::paintGL()
     // Draw bounding boxes
     if(m_drawBoundingBoxes && !m_incrementalDraw)
     {
-        for(size_t i = 0; i < geoms.size(); ++i)
-            drawBoundingBox(transState, geoms[i]->boundingBox(), Imath::C3f(1));
+        if (m_boundingBoxShader->isValid())
+        {
+            QGLShaderProgram &boundingBoxShader = m_boundingBoxShader->shaderProgram();
+            // shader
+            boundingBoxShader.bind();
+            // matrix stack
+            transState.setUniforms(boundingBoxShader.programId());
+
+            for (size_t i = 0; i < geoms.size(); ++i)
+            {
+                drawBoundingBox(transState, geoms[i]->getVAO("boundingbox"), geoms[i]->boundingBox().min,
+                                Imath::C3f(1), geoms[i]->shaderId("boundingbox")); //boundingBoxShader.programId()
+            }
+
+            // boundingBoxShader.release();
+        }
     }
 
     // Draw meshes and lines
     if (!m_incrementalDraw)
     {
-        drawMeshes(transState, geoms);
-
+        this->drawMeshes(transState, geoms);
         // Generic draw for any other geometry
         // (TODO: make all geometries use this interface, or something similar)
         // FIXME - Do generic quality scaling
         const double quality = 1;
         for (size_t i = 0; i < geoms.size(); ++i)
+        {
             geoms[i]->draw(transState, quality);
+        }
```

Comment:
Another style nitpick: the codebase is fairly consistent about not using extra brackets around single line loops or if statements.  Could you kill these off?